### ISSUE TITLE
[Bug]: rms_norm ignores a stacked per-row weight — vLLM DFlash _normalize_context_k applies layer 0's k_norm to every draft layer on XPU

### ISSUE THREAD

**Issue by martyrant** [vllm-project/vllm-xpu-kernels#573]
# [Bug]: XPU `rms_norm` ignores a stacked per-row weight — vLLM's DFlash `_normalize_context_k` applies layer 0's k_norm to every draft layer on XPU

## Environment

```
vllm-xpu-kernels 0.1.14.1 ; vLLM 0.28.1rc1.dev278+g73029d424 (vllm/vllm-openai-xpu nightly 2026-09-02) ; torch 2.13.0+xpu
Intel Arc Pro B70 (32 GB), Level Zero driver 26.27.39122.11, IGC 2.38.2 ; Ubuntu 24.04, kernel 7.0, xe
```

## Describe the bug

vLLM's DFlash/DFlash2 draft path (`vllm/model_executor/models/qwen3_dflash.py`, git sha 73029d424, `_normalize_context_k`; call-site lines 482 (`self._k_norm_weights = torch.stack(...)`), 560–569 (`_normalize_context_k` → `ops.rms_norm(all_k_normed, all_k, self._k_norm_weights, eps)`)) -> torch.Tensor: 564:        ops.rms_norm( ) normalizes the context K of all draft layers at once by calling `ops.rms_norm(out, all_k, self._k_norm_weights, eps)` with `all_k` shaped `[L, num_ctx, nkv, hd]` and a **stacked weight `[L, hd]`**, with the comment "The weight is selected per layer by the outermost (layer) index." That relies on the CUDA kernel's 2-D weight support (`weight_stride`, row chosen by the outermost index).

The XPU kernel (`csrc/layernorm.cpp`, `rms_norm`) derives `hidden_size = input.size(-1)`, `num_tokens = input.numel() / hidden_size`, only checks `weight->is_contiguous()`, and indexes the weight as a flat `[hidden_size]` array (`weight[idx]` / `v_w[idx]`). So on XPU **every layer's K is normalized with layer 0's k_norm weight**; the remaining L-1 rows of the weight are never read. This is dtype-independent and silently degrades DFlash/DFlash2 draft acceptance on XPU (bf16 included).

## Reproduction (direct kernel test, run on the B70 inside the nightly container)

```python
import torch
from vllm import _custom_ops as ops
dev="xpu"; dt=torch.float16; L,N,nkv,hd=2,4,8,128; eps=1e-6
x=torch.randn(L,N,nkv,hd,device=dev,dtype=dt)
w=torch.ones(L,hd,device=dev,dtype=dt); w[1]=0.0        # layer-1 weight all zeros -> per-row kernel must zero layer 1's output
out=torch.empty_like(x); ops.rms_norm(out,x,w,eps)
ref=x.float()*torch.rsqrt(x.float().pow(2).mean(-1,keepdim=True)+eps)   # = layer-0 (weight 1) result
print((out[0].float()-ref[0]).abs().max())   # 0.0010  layer 0 correct
print(out[1].float().abs().max())            # 3.793   layer 1 NOT zeroed
print((out[1].float()-ref[1]).abs().max())   # 0.0010  layer 1 == result with layer 0's weight
```

Expected (CUDA semantics the call site assumes): `out[1]` is all zeros. Observed on XPU: `out[1]` equals the layer-0-weight result.

## Suggested fix

Either add 2-D weight support to the XPU `rms_norm` (weight stride = hidden_size, row = token_idx / (numel_per_row)), matching `csrc/libtorch_stable/layernorm_kernels.cu` in vLLM, or have the vLLM call site fall back to a per-layer loop when `weight.dim() == 2` on platforms whose kernel lacks it. A shape check that rejects a 2-D weight would at least turn the silent mismatch into an error.

Related: vllm-project/vllm#55250 (DFlash2 fp16 0% acceptance on XPU; separate cause — NaN inside the draft decoder layers — but this bug affects the same path in every dtype).

**Comment by jikunshang** [comment]
cc @zufangzhu  @mayuyuace  @ccrhx4  can you take a look?

**Comment by ccrhx4** [comment]
cread https://github.com/vllm-project/vllm-xpu-kernels/pull/579 for the fix
